examples/multihead_param_groups.py

```diff
@@ -0,0 +1,94 @@
+################################################################################
+# Copyright (c) 2021 ContinualAI.                                              #
+# Copyrights licensed under the MIT License.                                   #
+# See the accompanying LICENSE file for terms.                                 #
+#                                                                              #
+# Date: 01-12-2020                                                             #
+# Author(s): Andrea Cossu                                                      #
+# E-mail: contact@continualai.org                                              #
+# Website: avalanche.continualai.org                                           #
+################################################################################
+
+"""
+This example trains a Multi-head model on Split MNIST with Elastich Weight
+Consolidation. Each experience has a different task label, which is used at test
+time to select the appropriate head.
+"""
+
+import argparse
+import torch
+from torch.nn import CrossEntropyLoss
+from torch.optim import SGD
+
+from avalanche.benchmarks.classic import SplitMNIST
+from avalanche.models import MTSimpleMLP
+from avalanche.training.supervised import EWC
+from avalanche.evaluation.metrics import forgetting_metrics, accuracy_metrics
+from avalanche.logging import InteractiveLogger
+from avalanche.training.plugins import EvaluationPlugin
+
+
+def main(args):
+    # Config
+    device = torch.device(
+        f"cuda:{args.cuda}" if torch.cuda.is_available() and args.cuda >= 0 else "cpu"
+    )
+    # model
+    model = MTSimpleMLP()
+
+    # CL Benchmark Creation
+    benchmark = SplitMNIST(n_experiences=5, return_task_id=True)
+    train_stream = benchmark.train_stream
+    test_stream = benchmark.test_stream
+
+    # Prepare for training & testing
+    g1 = {"params": [], "lr": 0.1}
+    g2 = {"params": [], "lr": 0.01}
+
+    for n, p in model.named_parameters():
+        if "classifier" in n:
+            g1["params"].append(p)
+        else:
+            g2["params"].append(p)
+
+    optimizer = SGD([g1, g2])
+    criterion = CrossEntropyLoss()
+
+    # choose some metrics and evaluation method
+    interactive_logger = InteractiveLogger()
+
+    eval_plugin = EvaluationPlugin(
+        accuracy_metrics(minibatch=False, epoch=True, experience=True, stream=True),
+        forgetting_metrics(experience=True),
+        loggers=[interactive_logger],
+    )
+
+    # Choose a CL strategy
+    strategy = EWC(
+        model=model,
+        optimizer=optimizer,
+        criterion=criterion,
+        train_mb_size=128,
+        train_epochs=3,
+        eval_mb_size=128,
+        device=device,
+        evaluator=eval_plugin,
+        ewc_lambda=0.4,
+    )
+
+    # train and test loop
+    for train_task in train_stream:
+        strategy.train(train_task, num_workers=4)
+        strategy.eval(test_stream, num_workers=4)
```

Comment:
can you print what's happening here? otherwise it's not very useful as an example.

Reply:
I could, but that would require some kind of verbose option in the make_optimizer function. I don't know if it's necessary I think the verbose option in the update_optimizer is more for debugging purposes. Let me know what you think.

Reply:
I think it would be useful, even if the prints are minimal and not too nice (we can improve them later if necessary). Otherwise, it may be difficult to understand what's going on for a user.